miden-tx/src/host/mod.rs

```diff
@@ -126,18 +125,25 @@ impl<A: AdviceProvider, T: TransactionAuthenticator> TransactionHost<A, T> {
         &mut self,
         process: &S,
     ) -> Result<(), TransactionKernelError> {
-        //# => [ASSET, note_ptr]
-        let note_ptr: MemoryAddress = process
-            .get_stack_item(4)
-            .try_into()
-            .map_err(TransactionKernelError::MalformedNotePointer)?;
+        let stack = process.get_stack_state();
+        //# => [ASSET, note_ptr, num_of_assets, note_idx]
+
+        // this is a safe conversion because the note index (in theory) is always a u64
+        // in theory, the user can provide any note index. However, we already check in
+        // the kernel that the note index < MAX_OUTPUT_NOTES_PER_TX
+        let note_idx_u64 = stack[6].as_int();
```

Comment:
@bobbinth we could assume here that `note_idx` can always safely be converted to `usize`. We check in the kernel that the `note_idx` < `4096` = `MAX_OUTPUT_NOTES_PER_TX`.

But as an additional check, I added this if statement here